### ISSUE TITLE
Ideal membership

### ISSUE THREAD

**Issue by CarloSircana** [oscar-system/Oscar.jl#485]
The ideal membership test seems to be broken. Here I define an ideal and construct a monomial which, according to the Oscar function, belongs to the ideal.
If I then compute the ideal of the leading terms, it tells me that the monomial is not in the leading term ideal.

```
R, x = PolynomialRing(QQ, "x"=>1:5)
I = ideal(R, [x[1]*x[4]^2 - x[2]*x[5]^2, x[1]^3*x[3]^3 - x[2]^4*x[4]^2, x[2]*x[4]^8 - x[3]^3*x[5]^6, x[1]^2, x[2]^5, x[3]^6, x[4]^18, x[5]^10])
m = x[2]^3*x[4]^6*x[5]^2
m in I
J = leading_ideal(I)
m in J
```

By looking by hand at the Groebner basis, I would say that the problem is in the ideal membership test.

**Comment by thofma** [comment]
@tthsqe12 I heard you wanted to work on some Oscar bugs

**Comment by wdecker** [comment]
@ederc : I guess the problem is not the ideal membership function. There seems to be, however, a problem with the Groebner basis: The dp GB in Singular reads:

_[1]=x(1)^2
_[2]=x(1)*x(4)^2-x(2)*x(5)^2
_[3]=x(1)*x(2)*x(5)^2
_[4]=x(2)^5
_[5]=x(2)^2*x(5)^4
_[6]=x(2)^4*x(4)^2
_[7]=x(3)^6
_[8]=x(2)*x(4)^8-x(3)^3*x(5)^6
_[9]=x(5)^10
_[10]=x(2)^2*x(4)^6*x(5)^2-x(1)*x(3)^3*x(5)^6
_[11]=x(1)*x(3)^3*x(5)^8
_[12]=x(4)^18
_[13]=x(3)^3*x(4)^8*x(5)^8
_[14]=x(3)^3*x(4)^10*x(5)^6

Please see _[10]

In Oscar I get:

ulia> groebner_basis(I)
14-element Array{fmpq_mpoly,1}:
 x[1]^2
 x[1]*x[4]^2 - x[2]*x[5]^2
 x[1]*x[2]*x[5]^2
 x[2]^5
 x[2]^2*x[5]^4
 x[2]^4*x[4]^2
 x[3]^6
 x[2]*x[4]^8 - x[3]^3*x[5]^6
 x[5]^10
 -x[1]*x[3]^3*x[5]^6 + x[2]^2*x[4]^6*x[5]^2
 x[1]*x[3]^3*x[5]^8
 x[4]^18
 x[3]^3*x[4]^8*x[5]^8
 x[3]^3*x[4]^10*x[5]^6

and consequently x[1]*x[3]^3*x[5]^8 is in the "leading ideal" rather than x[2]^2*x[4]^6*x[5]^2 which divides m

**Comment by tthsqe12** [comment]
I am adding this to the queue.
My initial hunch is that the monomial ordering got lost in translation.

**Comment by ederc** [comment]
Should be fixed with #489: Problem is mainly what @wdecker printed: The gb stored "on the Oscar" side sorts the monomials in general not in the monomial order we computed the gb for (on the Singular side). Falsely leading_ideal() just took the leading monomials of I.gb.O and not of I.gb.S.

**Comment by thofma** [comment]
Fixed by #489.
